### PR TITLE
Update downloadsItemView data on every change

### LINKED ISSUE #1624: Issues in Saving MIME-based download preferences
Steps to reproduce:
1. Go to **Tools** > **Preferences** > **Advanced** > **Downloads**
2. Select a MIME type
3. Select **Save to Disk**
4. Select a download location
5. Click **Apply**

Actual:
Save to disk download type is saved, download location is not saved.

Expected:
Both save to disk download type and download location are saved.

https://youtu.be/IkY0MuHbFsc

### PR DESCRIPTION
**Pull request by fnkkio**
Fixes #1624 
Probably a better fix than #1625

### CODE REVIEW

File: src/ui/FilePathWidget.cpp [hunk 1]
```diff
@@ -79,9 +79,20 @@ void FilePathWidget::focusInEvent(QFocusEvent *event)
...
 	QWidget::focusInEvent(event);
 
+	m_focusInPath = getPath();
 	m_lineEditWidget->setFocus();
 }
 
+void FilePathWidget::focusOutEvent(QFocusEvent *event)
+{
+	Q_UNUSED(event)
+
```
Comment (Emdek) [inline src/ui/FilePathWidget.cpp]: Events should be passed to base class:
```
QWidget::focusOutEvent(event);
```

File: src/ui/FilePathWidget.cpp [hunk 2]
```diff
@@ -79,9 +79,20 @@ void FilePathWidget::focusInEvent(QFocusEvent *event)
...
+	if (getPath() != m_focusInPath)
```
Comment (Emdek) [inline src/ui/FilePathWidget.cpp]:
```
if (m_focusInPath != getPath())
```

File: src/ui/FilePathWidget.h [hunk 3]
```diff
@@ -73,6 +74,7 @@ protected slots:
 	QCompleter *m_completer;
 	QString m_filter;
 	OpenMode m_openMode;
+	QString m_focusInPath;
```
Comment (Emdek) [inline src/ui/FilePathWidget.h]: Variables should be grouped by type, also, it needs better name.

**Comment by fnkkio** [comment]
@Emdek updated.

File: src/ui/FilePathWidget.cpp (hunk 1 above)
Comment (fnkkio) [reply src/ui/FilePathWidget.cpp]: thanks

**Pull request merged**